tests/correctness/functions.rs

```diff
@@ -270,6 +270,48 @@ fn function_block_instances_save_state_per_instance() {
     assert_eq!(interface.f.i, 2);
     assert_eq!(interface.j.i, 7);
 }
+
+#[test]
+#[ignore = "https://github.com/PLC-lang/rusty/issues/530"]
+fn function_block_instances_save_outputs() {
```

Comment:
if i didn't miss anything this test case should work just fine
have to look into why there was no validation for the variable declaration

Reply:
> line 295: "var : DINT" var is a keyword, we should validate this, therefore we didn't generate a correct IR "%main_interface = type {}" missing type in struct "type { i32 }" would be correct

Oof, you're completely right, sorry.  I can send a follow-up to fix the test.  Is there already an issue to track this missing validation?

> line 304: "j(a := FALSE, b => k);" k is not initialized and will have the default value FALSE, this will lead to FALSE OR FALSE resulting in FALSE

Right, this is the entire point of the test.  According to the 61131-3, this behavior is wrong.  The `VAR_OUTPUT` should be persisted as part of the function-block's state.  Whatever variable is assigned (or not assigned) to the `VAR_OUTPUT` should not influence the value in any way.  See my comments in #530.

Reply:
I agree with @Rahix here, k should have gotten the value of b.
Regarding the validation, we purposefully ignore it in codegen tests at the moment for historical reasons (The diagnostician is not invoked during that stage)
We have a story that has been on hold for a while (because of false positives in validation IIRC) to kind of address this #324